src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1020,16 +1038,66 @@ protected function prepare_items_query( $prepared_args = array(), $request = nul
 		}
 
 		// Map to proper WP_Query orderby param.
-		if ( isset( $query_args['orderby'] ) && isset( $request['orderby'] ) ) {
+		if ( isset( $query_args['orderby'] ) ) {
 			$orderby_mappings = array(
 				'id'            => 'ID',
 				'include'       => 'post__in',
 				'slug'          => 'post_name',
 				'include_slugs' => 'post_name__in',
 			);
 
-			if ( isset( $orderby_mappings[ $request['orderby'] ] ) ) {
-				$query_args['orderby'] = $orderby_mappings[ $request['orderby'] ];
+			// The orderby query string parameter can look like ?orderby=menu_order, ?orderby[]=menu_order, or ?orderby[menu_order]=asc.
+			if ( rest_is_array( $query_args['orderby'] ) ) {
+				// An array: ?orderby[]=menu_order
+				// A fake array: ?orderby=menu_order,title
+				// A string: ?orderby=menu_order
+				// The latter two because rest_is_array() and rest_sanitize_array() pass scalars through wp_parse_list()
+				$orderby_mappings_in_request = array_intersect_key( $orderby_mappings, array_flip( $query_args['orderby'] ) );
+
+				$new_arg = array();
+				// Remaps values. Preserves value order.
+				foreach ( $query_args['orderby'] as $orderby ) {
+					if ( array_key_exists( $orderby, $orderby_mappings_in_request ) ) {
+						$new_arg[] = $orderby_mappings_in_request[ $orderby ];
+					} else {
+						$new_arg[] = $orderby;
+					}
+				}
+
+				// If ID is not set, try stabilizing the order. A client can see unexpected ordering when
+				// items are technically equal to one another (say all pages have a menu_order of 0), but would like to
+				// see a stable sort. This can especially help for pagination cases.
+				// See https://core.trac.wordpress.org/ticket/46294 and https://core.trac.wordpress.org/ticket/44349
+				if ( ! in_array( 'ID', $new_arg, true ) && ! in_array( 'relevance', $new_arg, true ) ) {
+					$new_arg[] = 'ID';
```

Comment:
Do we want to enforce a direction for this added 'ID' sort? Currently this will pick up whatever order direction the query has set.

Also I chose 'desc' below due to the API defaults, but I do wonder if `asc` is more useful.

Reply:
Reading the other issues, I think this needs to be id `asc` to paginate a tree view in a better way.